sympy/parsing/latex/lark/transformer.py

```diff
@@ -107,37 +113,74 @@ def group_curly_parentheses(self, tokens):
         return tokens[1]
 
     def eq(self, tokens):
-        return sympy.Eq(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Eq(tokens[0], tokens[2])
+        else:
+            return sympy.Eq(tokens[0], tokens[2], evaluate=False)
 
     def ne(self, tokens):
-        return sympy.Ne(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Ne(tokens[0], tokens[2])
+        else:
+            return sympy.Ne(tokens[0], tokens[2], evaluate=False)
 
     def lt(self, tokens):
-        return sympy.Lt(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Lt(tokens[0], tokens[2])
+        else:
+            return sympy.Lt(tokens[0], tokens[2], evaluate=False)
```

Comment:
There are definitely better ways to structure this rather than duplicating the lines like this everywhere.

Reply:
There are lots of ways to do this but one possibility is like described in gh-25620. Basically make a different object here instead of `sympy` that calls these functions with `evaluate=False`. That could be added just as a private object for now but in general it is something that would be useful for users.

Reply:
That sounds like a big change that would greatly expand the scope of this PR.

I think that, as a first approximation, this code is fine. We can always refactor it later.

Reply:
It doesn't need to be a big change. That is why I suggested adding it as something private (then it is not necessary right now to figure out all of the details but just to make it work here). Ultimately longer term that is most likely how we would want to have this be implemented.

Reply:
So, if I understood correctly, you want me to actually add the files `sympy/sandbox/evaluatefalse.py` and `sympy/sandbox/evaluatetrue.py` as @sylee957 suggested in [#25620 (comment)](https://github.com/sympy/sympy/issues/25620#issue-1878270284), and then use those functions here?

I fail to see how that reduces code duplication, since I would need to define every single function used in the transformer class twice: once in `evaluatefalse.py` and once in `evaluatetrue.py`.

Reply:
No, I was just suggesting to have a class like:
```python
class builder:
    def __init__(self, evaluate):
        self.evaluate = evaluate

    def op(self, Op, args):
        if self.evaluate:
            return Op(*args)
        else:
            return Op(*args, evaluate=False)

    def Lt(self, a, b):
        return self.op(sympy.Lt, [a, b])
```
This could just be added for now as a private feature in the parsing module to be used by `parse_latex`. It could in future be used by all the parsers and could also serve as a starting point for implementing the expression builder pattern more generally in future.

In any case if you don't want to do this then there are many very simple ways to avoid the duplication above such as just defining the `op` method on the transformer class.

Reply:
Anyway, I'd identify that the priority is to fix the ambiguity issue in https://github.com/sympy/sympy/pull/25798#discussion_r1363824305
than doing the refactory.
If https://github.com/sympy/sympy/pull/25798#discussion_r1363824305 is technically not elegant to resolve, I'd also find that as an evidence for argument about https://github.com/sympy/sympy/pull/25790

Reply:
> If [#25798 (comment)](https://github.com/sympy/sympy/pull/25798#discussion_r1363824305) is technically not elegant to resolve, I'd also find that as an evidence for argument about #25790

I see it the other way around. If the grammar for the parser has been designed in such a way that it cannot parse simple expressions correctly without evaluation then that is a design flaw in the parser. The proper fix is to fix the grammar rather than deciding not to support parsing unevaluated expressions.